### PR TITLE
Switch to Wiggins for VPS Scans

### PR DESCRIPTION
**Pull request by kitified**
Switches to using the Wiggins CLI as an "unmanaged plugin", in other words a binary that is simply executed by `fossa` with the same arguments it is provided.

Removes all logic related to interacting with FOSSA services for VPS scan initiation or completion, instead all that is now handled by Wiggins.

Note: wiggins isn't _quite_ done yet, but won't significantly change at this point.

### CODE REVIEW

File: src/App/Fossa/VPS/Scan.hs
Comment (scruffystuffs) [inline src/App/Fossa/VPS/Scan.hs:43]: We can simplify the logging call here, since we're no longer using `concurrently`.

```suggestion
vpsScan (BaseDir basedir) logSeverity overrideProject skipIprFlag licenseOnlyScan fileFilters apiOpts metadata binaryPaths = withLogger logSeverity $ do
```

File: src/App/Fossa/VPS/Scan.hs
Comment (scruffystuffs) [inline src/App/Fossa/VPS/Scan.hs:50]: Since we aren't using `concurrently` anymore, we can just use the current effect wrappers.  We can remove the entire `runIt` line, and this can probably become
```haskell
wigginsResult <- sendIO . runExecIO $ runWiggins binaryPaths wigginsOpts
```

File: .github/workflows/build.yml
Comment (scruffystuffs) [inline .github/workflows/build.yml:86]: REALLY happy to see a simplified build process. 👍

File: src/App/Fossa/VPS/Scan/Core.hs
Comment (scruffystuffs) [inline src/App/Fossa/VPS/Scan/Core.hs:14]: This comment is no longer relevant, which is just great.  Please take great pleasure in removing it.

File: src/App/Fossa/VPS/Scan/RunWiggins.hs
Comment (scruffystuffs) [inline src/App/Fossa/VPS/Scan/RunWiggins.hs:55]: [nit]: This seems a little simpler as a list literal.

```suggestion
  [ "analyze",
    "-endpoint", renderUri apiOptsUri,
    "-fossa-api-key", unApiKey apiOptsApiKey,
    optMaybeText "-name" overrideName,
    optMaybeText "-revision" overrideRevision,
    optMaybeText "-jira-project-key" projectJiraKey,
    optMaybeText "-link" projectLink,
    optMaybeText "-policy" projectPolicy,
    optMaybeText "-project-url" projectUrl,
    optMaybeText "-team" projectTeam,
    optMaybeText "-title" projectTitle,
    optBool "-license-only" scanLicenseOnly,
    optBool "-skip-ipr-scan" scanSkipIpr,
    optBool "-debug" (logSeverity == SevDebug),
    optFilterExpressions fileFilters,
    "."
  ]
```

File: src/App/Fossa/VPS/Scan/Core.hs
Comment (scruffystuffs) [inline src/App/Fossa/VPS/Scan/Core.hs:57]: Since we removed the functions that this comment was referring to (except one), we can remove the comment as well.

**Review by scruffystuffs — CHANGES_REQUESTED** [review]
Structurally looks solid to me.  The important changes are the logging/effect bits, and removing now-useless comments.

File: src/App/Fossa/VPS/Scan/RunWiggins.hs
Comment (kitified) [reply src/App/Fossa/VPS/Scan/RunWiggins.hs]: This doesn't seem to work given that we are evaluating type `[Text]` from these `opt*` expressions. Am I missing something?

File: src/App/Fossa/VPS/Scan.hs
Comment (kitified) [reply src/App/Fossa/VPS/Scan.hs]: This doesn't seem to result in a `(Right, Left)` tuple, which I think we need for rendering the failure right?

File: src/App/Fossa/VPS/Scan/RunWiggins.hs
Comment (scruffystuffs) [reply src/App/Fossa/VPS/Scan/RunWiggins.hs]: Ah, that's right.  I think there's a way to make this more readable, but it might take some thinking and I'd rather not block on it, it was a nit anyway.

File: src/App/Fossa/VPS/Scan.hs
Comment (scruffystuffs) [reply src/App/Fossa/VPS/Scan.hs]: Any errors in the execution should be "thrown" when we run the effect at the `runExecIO` call.  That will raise any errors into the `Diagnostics` effect we're currently working in (this function has the constraint `Has Diagnostics sig m`), and therefore has to be handled by the caller.  Since `runWiggins` actually returns `m ()`, this line could probably just be `runExecIO $ runWiggins binaryPaths wigginsOpts`, without `sendIO` or `result <-`.  Actually throwing the error should be lower in the call chain, probably in `execWiggins`, but reporting it to console is done in `scanMain` in this file.

Comment (kitified) [reply src/App/Fossa/VPS/Scan.hs]: Ok, that makes sense. Made this change and the one in your other outstanding comment and both error and non-error cases work great! Thanks!

File: src/App/Fossa/VPS/Scan.hs
Comment (kitified) [reply src/App/Fossa/VPS/Scan.hs]: Made this change and the one in your other outstanding comment and both error and non-error cases work great! Thanks!

**Review by scruffystuffs — APPROVED** [review]
I'm cool with this, but I'd personally prefer if @cnr reviewed as well before merging.

**Review by cnr — APPROVED** [review]
This looks great

Please make sure to resolve the endpoint-trailing-slash issue in wiggins; it feels like something that will bite us eventually if left unresolved

**Review by spatten — APPROVED** [review]
This looks good to me as well. So much deleted code!

**Comment by kitified** [comment]
> This looks great
> 
> Please make sure to resolve the endpoint-trailing-slash issue in wiggins; it feels like something that will bite us eventually if left unresolved

This is actually resolved at this point in wiggins, thanks for the reminder- removed the workaround from spectrometer.

**Comment by kitified** [comment]
Final wiggins changes are landed, merging!

**Pull request merged**
